demos/generomak_solps_from_raw_files.py

```diff
@@ -0,0 +1,234 @@
+    sys.exit(1)
+
+###############################################################################
+# Create a Cherab plasma from the simulation
+###############################################################################
+mesh = sim.mesh
+plasma = sim.create_plasma()
+
+me = mesh.mesh_extent
+xl, xu = (me['minr'], me['maxr'])
+zl, zu = (me['minz'], me['maxz'])
+nx = 500
+ny = 500
+
+# Extract information about the species in the plasma.
+h0 = plasma.composition.get(hydrogen, 0)
+h1 = plasma.composition.get(hydrogen, 1)
+c0 = plasma.composition.get(carbon, 0)
+c1 = plasma.composition.get(carbon, 1)
+c2 = plasma.composition.get(carbon, 2)
+c3 = plasma.composition.get(carbon, 3)
+c4 = plasma.composition.get(carbon, 4)
+c5 = plasma.composition.get(carbon, 5)
+c6 = plasma.composition.get(carbon, 6)
+
+###############################################################################
+# Sample some poloidal profiles from the simulation.
+###############################################################################
+print('Sampling profiles...')
+# The distributions are 3D, so we perform a sample in 3D space with only a
+# single y value to get a poloidal profile.
+xsamp, _, zsamp, ne_samples = sample3d(plasma.electron_distribution.density,
+                                       (xl, xu, nx), (0, 0, 1), (zl, zu, ny))
+ne_samples = ne_samples.squeeze()
+te_samples = sample3d_grid(plasma.electron_distribution.effective_temperature,
+                           xsamp, [0], zsamp).squeeze()
+h0_samples = sample3d_grid(h0.distribution.density, xsamp, [0], zsamp).squeeze()
+h1_samples = sample3d_grid(h1.distribution.density, xsamp, [0], zsamp).squeeze()
+c0_samples = sample3d_grid(c0.distribution.density, xsamp, [0], zsamp).squeeze()
+c1_samples = sample3d_grid(c1.distribution.density, xsamp, [0], zsamp).squeeze()
+c2_samples = sample3d_grid(c2.distribution.density, xsamp, [0], zsamp).squeeze()
+c3_samples = sample3d_grid(c3.distribution.density, xsamp, [0], zsamp).squeeze()
+c4_samples = sample3d_grid(c4.distribution.density, xsamp, [0], zsamp).squeeze()
+c5_samples = sample3d_grid(c5.distribution.density, xsamp, [0], zsamp).squeeze()
+c6_samples = sample3d_grid(c6.distribution.density, xsamp, [0], zsamp).squeeze()
+# Magnitude of main ion velocity vector.
+h1_velocity = samplevector3d_grid(h1.distribution.bulk_velocity, xsamp, [0], zsamp).squeeze()
+h1_speed = linalg.norm(h1_velocity, axis=-1)
+# Mask determining whether a point is inside or outside the simulation mesh.
+# This is a 2D function: see sim.inside_volume_mesh for the 3D equivalent.
+inside_samples = sample2d_grid(sim.inside_mesh, xsamp, zsamp)
+
+# Plot the simulation mesh.
+plt.ion()
+mesh.plot_mesh()
+plt.title('Mesh geometry')
+
+# Plot sampled values.
+plt.figure()
+plt.imshow(ne_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('Electron density [m-3]')
```

Comment:
This is very optional, but the units can be shown in the colorbar instead of the title.
```suggestion
plt.colorbar(label='m$^{-3}$')
plt.xlim(xl, xu)
plt.ylim(zl, zu)
plt.title('Electron density')
```

Reply:
If this were for a publication I'd agree with you. For a simple code demo I don't think it's as important. I've just tried both ways and I prefer being able to quickly see the units right next to the quantity in the title rather than having to look half way down the colour bar for them.